src/subbot/DN/STGReq.py

```diff
@@ -18,27 +20,60 @@ def __init__(self, bot: commands.Bot) -> None:
             'mdef': ['Pdef/Mdef Cap', 'Pdef/Mdef Penalty', 'Pdef/Mdef % Cap'],
             'fd': ['FD Cap', 'FD Penalty', 'FD % Cap'],
         }
+        self.stgreq_command_ordered = ['crit', 'cdm', 'def', 'fd']
+        self.stgreq_command_long = {
+            'cdmg': 'Crit. Damage',
+            'cdm': 'Crit. Damage',
+            'crit': 'Critical',
+            'def': 'Pdef/Mdef',
+            'mdef': 'Pdef/Mdef',
+            'fd': 'Final Damage',
+        }
+        self.logger = setup_logger("DNSTGReq")
 
     @commands.hybrid_command(name='stgreq', aliases=['crit', 'cdmg', 'cdm', 'def', 'mdef', 'fd'],
-                             help='Returns the required stats given the stg requirement (13-20). ' +
+                             help='Returns the required stats given the stg requirement (13-25). ' +
                                   'Refer to the command alias for stats.')
     async def stgreq(self, ctx: commands.Context, stg: int) -> None:
-
-        stat = self.command_mapper.get(ctx.invoked_with, self.df.columns)
-
         # Return error message if stg is not within range
         if stg not in range(self.stg_min, self.stg_max + 1):
-            res = f"STG argument is wrong! Please enter a valid integer between {self.stg_min} and {self.stg_max}."
+            message = f"STG argument is wrong! Please enter a valid integer between {self.stg_min} and {self.stg_max}."
+            await ctx.send(message)
+            return
 
-        # Return the stat for a given stg
+        # Get set of stats to
+        stats = list()
+        if ctx.invoked_with.lower() == "stgreq":
+            stats.extend(self.stgreq_command_ordered)
         else:
-            res = self.df[stat]
-            if stg:
-                res = res.loc[stg]
-            res = res.to_string()
+            stats.append(ctx.invoked_with)
+
+        # Generate and fill the output table
+        res = []
+        print(stats)
+
+        for idx, stat in enumerate(stats):
+            columns = self.command_mapper.get(stat, self.df.columns)
+            print(f"columns to process are {columns}")
+            # Return the stat for a given stg
+            data = self.df[columns]
+            data = data.loc[stg]
+            print(data)
+            entry = {
+                "Index": idx,
+                "Stat": self.stgreq_command_long[stat].title(),
+                "Cap": data[0],
+                "Penalty": data[1],
+                "% Cap": data[2]
+            }
+            res.append(entry)
```

Comment:
Could be a `map` if you want.

Reply:
If don't want, please create issue instead.

Reply:
Hmm isn't this a dictionary alrd? Or am I misunderstanding something?

Reply:
Sorry, the `map()` function

Reply:
nvm this